db/document.go

```diff
@@ -971,7 +1094,8 @@ func (doc *Document) UnmarshalWithXattr(data []byte, xdata []byte, unmarshalLeve
 	if len(data) == 0 && len(xdata) > 0 {
 		doc._body = Body{}
 		doc._rawBody = []byte(base.EmptyDocument)
-		doc.Deleted = true
+		//doc.Deleted = true
+		doc._body[BodyDeleted] = true
```

Comment:
We should be setting the deleted flag, not inserting another special property to deal with.